CONTRIBUTING.md

```diff
@@ -56,14 +36,138 @@ When creating your enhancement request, please:
+
+    </details>
+
+4. **Test your changes**
+
+    <details><summary>Expand details 👇</summary><br/>
+
+    Our continuous integration (CI) testing runs [a number of checks](https://github.com/allenai/allennlp/actions?query=workflow%3APR) for each pull request on [GitHub Actions](https://github.com/features/actions). You can run most of these tests locally, which is something you should do *before* opening a PR to help speed up the review process and make it easier for us.
+    
+    First, you should run [`black`](https://github.com/psf/black) to make sure you code is formatted consistently. Many IDEs support code formatters as plugins, so you may be able to setup black to run automatically everytime you save. [`black.vim`](https://github.com/psf/black/tree/master/plugin) will give you this functionality in Vim, for example. But `black` is also easy to run directly from the command line. Just run this from the root of your clone:
+    
+        black .
+
+    Our CI also uses [`flake8`](https://github.com/allenai/allennlp/tree/master/tests) to lint the code base and [`mypy`](http://mypy-lang.org/) for type-checking. You should run both of these next with
+
+        make lint
+
+    and
+
+        make typecheck
+
+    We also strive to maintain high test coverage, so most contributions should include additions to [the unit tests](https://github.com/allenai/allennlp/tree/master/tests). These tests are run with [`pytest`](https://docs.pytest.org/en/latest/), which you can use to locally run any test modules that you've added or changed.
+
+    For example, if you've fixed a bug in `allennlp/nn/util.py`, you can run the tests specific to that module with
+    
+        pytest -v tests/nn/util_test.py
+    
+    Our CI will automatically check that test coverage stays above a certain threshold (around 90%). To check the coverage locally in this example, you could run
+    
+        pytest -v --cov allennlp.nn.util tests/nn/util_test.py
+
+     If your contribution involves changes to any docstrings, you should make sure the API documentation can build without errors. For that, just run
+
+        make build-docs
+
+    If the build fails, it's most likely due to small formatting issues. If the error message isn't clear, feel free to comment on this in your pull request.
+
+    You can also serve and view the docs locally with
+    
+        make serve-docs
+
+    And finally, please update the [CHANGELOG](https://github.com/allenai/allennlp/blob/master/CHANGELOG.md) with notes on your contribution in the "Unreleased" section at the top.
+
+    After all of the above checks have passed, you can now open [a new GitHub pull request](https://github.com/allenai/allennlp/pulls).
+    Make sure you have a clear description of the problem and the solution, and include a link to relevant issues.
+
+    We look forward to reviewing your PR!
+
+    </details>
+
+### New models
+
+**Do you have a new state-of-the-art model?**
```

Comment:
@matt-gardner you may want to take another look at this now

Reply:
Sure, seems ok.  I think eventually we'll want a better story here, but we don't currently have one, so this is good enough.